analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ClassShouldNotBeEmpty.cs

```diff
@@ -73,6 +75,13 @@ class GenericNotEmptyWithConstraints<T>
     void Method(T arg) { }
 }
 
+class IntegerList: List<int> { }             // Compliant - creates a more specific type without adding new members to it (similar to using the typedef keyword in C/C++)
+class StringLookup<T>: Dictionary<string, T> { }
+interface IIntegerSet<T>: ISet<int> { }
+
+[ComVisible(true)]
+class ClassWithAttribute { }                 // Compliant - types with attributes are ignored
+
```

Comment:
maybe also programming error, to have count zero

```
    [] // Error
    public void M() {
    }
```

Reply:
Added it. In VB.NET it still recognizes the empty class. It seems the empty attribute list doesn't confuse the compiler there.